indico/modules/events/registration/fields/simple.py

```diff
@@ -391,41 +391,31 @@ def default_value(self):
 
     @property
     def view_data(self):
-        available_places = self._get_max_persons()
-        if available_places is not None:
-            available_places -= self.get_places_used()
-            if available_places < 0:
-                available_places = 0
-        return dict(super().view_data, available_places=available_places)
+        return dict(super().view_data, available_places=self.get_available_places(True))
 
     def get_validators(self, existing_registration):
         def _check_number_of_places(new_data):
             if existing_registration:
                 old_data = existing_registration.data_by_field.get(self.form_item.id)
                 if not old_data or not self.has_data_changed(new_data, old_data):
                     return True
-            if new_data and (max_persons := self._get_max_persons()):
-                places_left = max_persons - self.get_places_used()
-                if not existing_registration and self.form_item.data.get('persons_count_against_limit'):
-                    # TODO remove this once get_places_used's TODO is done
-                    places_left -= 1
-                if new_data and not places_left:
-                    raise ValidationError(_('There are no places left for this option.'))
+            if (new_data
```

Comment:
Maybe this logic deserves a unit test... some cases to cover:

- new registration
- updating registration, field untouched
- updating registration, field touched but same/lower count
- updating registration, field touched and count increased (this should fail if limit exceeded)